src/Bicep.Core/Intermediate/Expression.cs

```diff
@@ -492,13 +492,13 @@ public record DeclaredResourceExpression(
     SyntaxBase BodySyntax,
     Expression Body,
     ImmutableArray<ResourceDependencyExpression> DependsOn,
-    Expression? Description = null,
-    ArrayExpression? RetryOn = null,
-    ArrayExpression? WaitUntil = null
+    Expression? Description = null
```

Comment:
FYI making any changes here will also require changing:
Expression visitor:
https://github.com/Azure/bicep/blob/80a33d197d3fa9d51ddd76f913d47063ba1465cd/src/Bicep.Core/Intermediate/ExpressionVisitor.cs#L202-L204
Expression rewrite visitor:
https://github.com/Azure/bicep/blob/80a33d197d3fa9d51ddd76f913d47063ba1465cd/src/Bicep.Core/Intermediate/ExpressionRewriteVisitor.cs#L380-L389

I think this was missed in the original PR which added RetryOn and WaitUntil, which is why you didn't get a build break.